tests/platform_tests/test_idle_driver.py

```diff
@@ -0,0 +1,25 @@
+"""
+Some devices have potential problems entering idle state. So
+on m0 devices, we expect to disable both intel idle driver
+and acpi idle driver, and no available idle state for all cpu.
+"""
+import logging
+import pytest
+
+from tests.common.helpers.assertions import pytest_assert
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology('m0'),
+]
+
+def test_idle_driver(duthosts, enum_rand_one_per_hwsku_hostname):
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+    is_intel = 'Intel' in duthost.shell('lscpu | grep Vendor')['stdout']
```

Comment:
AMD CPU also have c state defined. 
By checking this thread, we can use same approach to check:
https://askubuntu.com/questions/1242503/how-do-you-check-if-processor-max-cstate-is-set-correctly-using-terminal-tilix

Reply:
If we do not have the same device hang problem on amd cpu, do we need to worry about it?